docs/deployment/README.md

```diff
@@ -52,13 +52,13 @@ For other cases,
 are available.
 
 ```shell
- $ tfenv install 0.12.12
-[INFO] Installing Terraform v0.12.12
+ $ tfenv install 0.14.4
+[INFO] Installing Terraform v0.14.4
```

Comment:
From the ticket it looks like the intention here is to use features available in Terraform v0.13 and greater (and that's what we're now requiring in our modules below). I'm wondering if it would make more sense to instruct users to use 0.13.6 (last patch version of 0.13) here instead of the latest version, 0.14.4?

My concern is mainly around having users on the same page. There shouldn't be breaking changes between minor versions so technically 0.13 and 0.14 should be interchangeable but I'm not sure if there would be noticeable differences that could confuse users. 

Or, if we're mostly testing this with 0.14.4, maybe the required version should be >=0.14?

Reply:
From what I read, the ticket says that they want to be able to use more recent versions of Terraform. At the time of writing, 0.14 was only in beta, but now it has stable releases so it seems like we should upgrade to support it, if possible.

And indeed, in my testing, there are only breaking changes between 0.12 and 0.13, not 0.13 and 0.14. Most of the changes in this PR involve updating our Terraform code for compatibility with > 0.13. 

I'm not sure if that answers the question?

Reply:
I take it all back. We had to make the upgrade only to 0.13.6 for now because for various reasons, you can't easily upgrade from 0.12.12 straight to 0.14.4